### ISSUE TITLE
Rename tree.length attribute to tree.span?

### ISSUE THREAD

**Issue by hyanwong** [tskit-dev/msprime#662]
At the moment the msprime/tskit API defines `SparseTree.length` to be the genomic length covered by a tree. It can be quite confusing when branches on a tree can have "lengths" too. I’ve been trying to use the word "span" instead of "length" to refer to the genome span covered by an edge or a tree. Would it be sensible to add the attribute .span as a preferred alternative to .length on a SparseTree (and perhaps change the description in other places in the documentation where we refer to extent over a genomic region, such as the "span" (i.e. length) of an edge).

**Comment by jeromekelleher** [comment]
I like this --- having "length" refer to two different things is confusing. However, we still have the ``sequence_length`` attribute of the tree sequence itself, which we can't rename without a great deal of pain. Having ``span`` for tree seems like a good idea to me though.

**Comment by petrelharp** [comment]
Yes, good call. So, would `SparseTree.length` be deprecated?

**Comment by jeromekelleher** [comment]
Yeah, I think so. I'd imagine keeping it around for a long time though, no point in breaking people's code.
